drivers/syslog/ramlog.c

```diff
@@ -290,10 +274,16 @@ static ssize_t ramlog_addbuf(FAR struct ramlog_dev_s *priv,
   flags = enter_critical_section();
 
 #ifdef CONFIG_RAMLOG_SYSLOG
-  if (header->rl_magic != RAMLOG_MAGIC_NUMBER && priv == &g_sysdev)
+  if (priv == &g_sysdev && header == NULL)
     {
+      priv->rl_header = (FAR void *)g_sysbuffer;
```

Comment:
what's the benefit to change the static data initialization to the runtime initialization?